### PR TITLE
crimson/osd: add osd to crush when it boots

### PR DESCRIPTION
**Pull request by tchaikov**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/crimson/osd/osd.cc [hunk 1]
```diff
@@ -285,6 +287,40 @@ seastar::future<> OSD::_send_boot()
...
+    "prefix": "osd crush create-or-move",
+    "id": {},
+    "weight": {:.4f},
+    "args": [{}]
+  }})", whoami, weight, loc);
+  return monc->run_command({cmd}, {}).then(
+    [this](int32_t code, string message, bufferlist) {
+      if (code) {
+	logger().warn("fail to add to crush: {} ({})", message, code);
+	throw std::runtime_error("fail to add to crush");
```
Comment (tchaikov) [inline src/crimson/osd/osd.cc]: i understand this is not *runtime_error*. will try to find/create a better exception type for this purpose.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-06-21_15:28:26-rados-wip-kefu-testing-2019-06-21-2037-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-06-22_17:53:44-rados-wip-kefu-testing-2019-06-21-2037-distro-basic-smithi/

**Comment by tchaikov** [comment]
@athanatos @rzarzynski ping?

**Comment by athanatos** [comment]
LGTM

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-06-24_01:13:41-rados-wip-kefu-testing-2019-06-23-2322-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-06-25_00:46:45-rados-wip-kefu-testing-2019-06-23-2322-distro-basic-smithi/

the failure are tracked by http://tracker.ceph.com/issues/40533

**Pull request merged**